File: VB/DxBlazorApplication1/Services/Interfaces/IStatusService.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DxBlazorApplication1.Services.Interfaces

    Public Interface IStatusService(Of TStatus)

        Public Function GetStatusesAsync() As Task(Of IList(Of TStatus))

    End Interface
End Namespace

File: CS/DxBlazorApplication1/Services/Interfaces/IStatusService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DxBlazorApplication1.Services.Interfaces {
    public interface IStatusService<TStatus> {
        public Task<IList<TStatus>> GetStatusesAsync();
    }
}
